### ISSUE TITLE
If the device settings cache gets stale with a non-existent facility, Kolibri does not recover

### ISSUE THREAD

**Issue by rtibbles** [learningequality/kolibri#8435]
## Observed behavior
If the Facility that is stored in the device settings cache as the default facility has been deleted, multiple places that attempt to access this information can fall into unrecoverable error states.

## Errors and logs

```
ERROR 2021-09-14 15:42:23,096 django.request Internal Server Error: /api/auth/session/current/
Traceback (most recent call last):
  File "/home/kivy/build/android/kolibri/dist/django/db/models/fields/related_descriptors.py", line 178, in __get__
AttributeError: 'DeviceSettings' object has no attribute '_default_facility_cache'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/kivy/build/android/kolibri/dist/django/core/handlers/exception.py", line 41, in inner
  File "/home/kivy/build/android/kolibri/dist/django/core/handlers/base.py", line 187, in _get_response
  File "/home/kivy/build/android/kolibri/dist/django/core/handlers/base.py", line 185, in _get_response
  File "/home/kivy/build/android/kolibri/dist/django/views/decorators/csrf.py", line 58, in wrapped_view
  File "/home/kivy/build/android/kolibri/dist/rest_framework/viewsets.py", line 116, in view
  File "/home/kivy/build/android/kolibri/dist/django/utils/decorators.py", line 67, in _wrapper
  File "/home/kivy/build/android/kolibri/dist/django/utils/decorators.py", line 149, in _wrapped_view
  File "/home/kivy/build/android/kolibri/dist/django/utils/decorators.py", line 63, in bound_func
  File "/home/kivy/build/android/kolibri/dist/rest_framework/views.py", line 495, in dispatch
  File "/home/kivy/build/android/kolibri/dist/rest_framework/views.py", line 455, in handle_exception
  File "/home/kivy/build/android/kolibri/dist/rest_framework/views.py", line 492, in dispatch
  File "/home/kivy/build/android/kolibri/core/auth/api.py", line 759, in update
  File "/home/kivy/build/android/kolibri/core/auth/api.py", line 765, in get_session_response
  File "/home/kivy/build/android/kolibri/core/auth/models.py", line 522, in session_data
  File "/home/kivy/build/android/kolibri/core/auth/models.py", line 1311, in get_default_facility
  File "/home/kivy/build/android/kolibri/core/device/utils.py", line 29, in get_device_setting
  File "/home/kivy/build/android/kolibri/dist/django/db/models/fields/related_descriptors.py", line 184, in __get__
  File "/home/kivy/build/android/kolibri/dist/django/db/models/fields/related_descriptors.py", line 159, in get_object
  File "/home/kivy/build/android/kolibri/dist/django/db/models/query.py", line 378, in get
kolibri.core.auth.models.DoesNotExist: Facility matching query does not exist.
```

## Expected behavior

Kolibri should catch this error, clear the stale cache, and recover.

## User-facing consequences
Seemingly permanent breakage of Kolibri.

## Steps to reproduce
Delete a facility, having already accessed the default facility from the Device Settings.

**Comment by ozer550** [comment]
Hi, I would like to work on this can you guide me on how to proceed.

**Comment by abhimnc** [comment]
@rtibbles  Please assign me this issue. I have two pull request merged #8782 and #8720  and want to tackle this issue this time.

**Comment by rtibbles** [comment]
Apologies @abhimnc - I had forgot to flag that this is resolved in the course of #8742
